### PR TITLE
Feature/pantry view

### LINKED ISSUE #59: POST Requests
- [x] Successful POST/GET requests
- [x] Unsuccessful POST/GET request message displayed to user
- [x] Validate input fields

### PR DESCRIPTION
**Pull request by mr-ryan12**
What (if any) features are you implementing?
- Allow a user to add ingredients to their pantry
- Update the view of the pantry
- Correctly calculate the number of ingredients needed
- Correctly calculate the number of ingredients removed once cooked

What (if anything) did you refactor?
- Refactored the domUpdates.js and scripts.js files

Were there any issues that arose?
- Not that weren't resolved

Is there anything that you need from your teammate?
- Pull request merge and code review

Any other comments, questions, or concerns?
- Not at the moment

### CODE REVIEW

File: src/classes/RecipeRepository.js [hunk 1]
```diff
@@ -54,6 +55,10 @@ class RecipeRepository {
       )
     }) : [];
   }
+
+  assignCurrentRecipe(recipe) {
+    this.currentRecipe = recipe;
+  }
```
Comment (philalewis) [inline src/classes/RecipeRepository.js:61]: Added method to assign a new property for use in the scripts file

File: src/classes/User.js [hunk 2]
```diff
@@ -24,6 +26,10 @@ class User {
     const index = this.cookbook.indexOf(recipe);
     this.cookbook.splice(index, 1);
   }
+
+  createPantry(pantry, data) {
+    return new Pantry(pantry, data)
```
Comment (philalewis) [inline src/classes/User.js:31]: Used a class instantiation of the panty class to assign the pantry

File: src/scripts.js [hunk 3]
```diff
@@ -207,3 +235,68 @@ function createCookbook() {
...
+    // const newIngredient = ingredients.find(ingredient => `${ingredient.id}` === input.id);
+    let newIngredient = user.pantry.ingredients.find(ingredient => {
+      return `${ingredient.id}` === input.id
+    })
+    if (!newIngredient) {
+      newIngredient = new Ingredient(parseInt(input.id), {amount: parseInt(input.value), unit: ''}, ingredientsData)
+      user.pantry.addIngredient(newIngredient)
+    }
+    if(parseInt(input.value) > 0) {
+      makePostRequest(data, newIngredient, parseInt(input.value));
```
Comment (philalewis) [inline src/scripts.js:268]: This took some time. Seems to be working well

File: src/scripts.js [hunk 4]
```diff
@@ -207,3 +235,68 @@ function createCookbook() {
...
+    }
+  })
+}
+
+function makePostRequest(data, currentIngredient, amount) {
+  updatePantry(data)
```
Comment (philalewis) [inline src/scripts.js:274]: Good helper function to use in other functions

File: src/styles.css [hunk 5]
```diff
@@ -244,14 +255,95 @@ h5 {
...
+  z-index: 1;
+  /* padding-top: 100px; */
+  left: 0;
+  top: 0;
+  width: 100%;
+  height: 100%;
+  overflow: auto;
+  background-color: rgb(0,0,0);
+  background-color: rgba(0,0,0,0.4);
+  display: flex;
```
Comment (philalewis) [inline src/styles.css:272]: I like this modal

**Review by philalewis — APPROVED** [review]

**Review by oacortez — COMMENTED** [review]
Thank you for adding additional code to our feature branch. I notice we resolved our issue where the user can remove and add ingredients to their pantry. I will be pushing this. Thank you!

**Pull request merged**